lib/ansible/module_utils/basic.py

```diff
@@ -1767,23 +1767,23 @@ def safe_eval(self, value, locals=None, include_exceptions=False):
                 return (value, e)
             return value
 
-    def _check_type_str(self, value):
+    def _check_type_str(self, key, value):
         if isinstance(value, string_types):
             return value
 
         # Ignore, warn, or error when converting to a string.
-        # The current default is to warn. Change this in Anisble 2.12 to error.
+        # The current default is to warn. Change this in Ansible 2.12 to error.
         common_msg = 'quote the entire value to ensure it does not change.'
         if self._string_conversion_action == 'error':
             msg = common_msg.capitalize()
             raise TypeError(msg)
         elif self._string_conversion_action == 'warn':
-            msg = ('The value {0!r} (type {0.__class__.__name__}) in a string field was converted to {1!r} (type string). '
-                   'If this does not look like what you expect, {2}').format(value, to_text(value), common_msg)
+            msg = ('The key {0!r} with value {1!r} (type {1.__class__.__name__}) in a string field was converted to {2!r} (type string). '
```

Comment:
I added `key` for this line as it was very confusing for me to understand which Argument Spec key is converted.